### PR TITLE
take snapshot at specified block and slightly better informants

### LINKED ISSUE #1857: Better informants for snapshots

### LINKED ISSUE #1859: Snapshot should be able to accept block number
Aside from defaulting to the tip - 1000, there should be an option to create the snapshot at a particular block number or block hash. e.g.

```
parity snapshot /tmp/2000000.snapshot --at 2000000
```

### PR DESCRIPTION
**Pull request by rphmeier**
Fixes #1859 
Fixes #1857 

This does fail immediately if the block is older than HISTORY and the database is pruned. I don't see any other course of action but feel free to recommend one.

### CODE REVIEW

File: parity/cli.rs [hunk 1]
```diff
@@ -231,6 +231,12 @@ Import/Export Options:
   --format FORMAT          For import/export in given format. FORMAT must be
                            one of 'hex' and 'binary'.
 
+Snapshot Options:
```
Comment (gavofyork) [inline parity/cli.rs]: is `latest` the latest - 1000?

Comment (rphmeier) [inline parity/cli.rs]: yep

File: parity/configuration.rs [hunk 2]
```diff
@@ -186,6 +187,7 @@ impl Configuration {
 				file_path: self.args.arg_file.clone(),
 				wal: wal,
 				kind: snapshot::Kind::Restore,
+				block_at: try!(to_block_id("latest")), // unimportant.
```
Comment (gavofyork) [inline parity/configuration.rs]: shouldn't even be here, really...

Comment (rphmeier) [inline parity/configuration.rs]: i can refactor the common fields out into a struct of its own, but then you end up with ~5 extra lines of boilerplate as opposed to one.

**Pull request merged**